src/plugins/intel_cpu/src/nodes/executors/shl/shl_eltwise.cpp

```diff
@@ -0,0 +1,226 @@
+                       static_cast<int>(shlEltwiseAttrs.algorithm));
+    }
+
+    return init_func != nullptr && init_func() == CSINN_TRUE;
+}
+
+void ShlEltwiseExecutor::exec(const std::vector<MemoryCPtr> &src,
+                              const std::vector<MemoryPtr> &dst,
+                              const void *post_ops_data_) {
+    for (size_t i = 0; i < src.size(); i++) {
+        srcTensors[i] = srcTensors[i].cloneWithNewShape(src[i]->getDescPtr()->getShape().getStaticDims());
+        srcTensors[i].setData(src[i]->getData());
+    }
+    for (size_t i = 0; i < dst.size(); i++) {
+        dstTensors[i] = dstTensors[i].cloneWithNewShape(dst[i]->getDescPtr()->getShape().getStaticDims());
+        dstTensors[i].setData(dst[i]->getData());
+    }
+
+    switch (shlEltwiseAttrs.algorithm) {
+    case Algorithm::EltwiseAdd:
+        setExecFunc(csinn_add, srcTensors[0].get(), srcTensors[1].get(), dstTensors[0].get(), static_cast<csinn_diso_params*>(params->get()));
+        break;
+    case Algorithm::EltwiseSubtract:
+        setExecFunc(csinn_sub, srcTensors[0].get(), srcTensors[1].get(), dstTensors[0].get(), static_cast<csinn_diso_params*>(params->get()));
+        break;
+    case Algorithm::EltwiseMultiply:
+        setExecFunc(csinn_mul, srcTensors[0].get(), srcTensors[1].get(), dstTensors[0].get(), static_cast<csinn_diso_params*>(params->get()));
+        break;
+    case Algorithm::EltwiseDivide:
+        setExecFunc(csinn_div, srcTensors[0].get(), srcTensors[1].get(), dstTensors[0].get(), static_cast<csinn_diso_params*>(params->get()));
+        break;
+    case Algorithm::EltwiseMaximum:
+        setExecFunc(csinn_maximum, srcTensors[0].get(), srcTensors[1].get(), dstTensors[0].get(), static_cast<csinn_diso_params*>(params->get()));
+        break;
+    case Algorithm::EltwiseMinimum:
+        setExecFunc(csinn_minimum, srcTensors[0].get(), srcTensors[1].get(), dstTensors[0].get(), static_cast<csinn_diso_params*>(params->get()));
+        break;
+    case Algorithm::EltwiseExp:
+        setExecFunc(csinn_exp, srcTensors[0].get(), dstTensors[0].get(), static_cast<csinn_siso_params*>(params->get()));
+        break;
+    case Algorithm::EltwiseRelu:
+        if (shlEltwiseAttrs.alpha == 0) {
+            setExecFunc(csinn_relu, srcTensors[0].get(), dstTensors[0].get(), static_cast<csinn_relu_params*>(params->get()));
+        }
+        else {
+            setExecFunc(csinn_leaky_relu, srcTensors[0].get(), dstTensors[0].get(), static_cast<csinn_relu_params*>(params->get()));
+        }
+        break;
+    case Algorithm::EltwisePrelu:
+        setExecFunc(csinn_prelu, srcTensors[0].get(), srcTensors[1].get(), dstTensors[0].get(), static_cast<csinn_prelu_params*>(params->get()));
+        break;
+    default:
+        OPENVINO_THROW("Unsupported operation type for SHL Eltwise executor: ",
+                       static_cast<int>(shlEltwiseAttrs.algorithm));
+    }
```

Comment:
What's about to hide these setters to `init` stage?
We already have `switch-case` in `init`. Can we init `exec_func` there?
```
class ShlEltwiseExecutor {
private:
     std::function<int()> exec_func;
};

bool ShlEltwiseExecutor::init(...) {
     switch (shlEltwiseAttrs.algorithm) {
     case Algorithm::EltwiseAdd:
           INIT(csinn_add_init(srcTensors[0].get(), srcTensors[1].get(), dstTensors[0].get(), static_cast<csinn_diso_params*>(params->get())));
           exec_func = [&]() {
            return csinn_add(srcTensors[0].get(), srcTensors[1].get(), dstTensors[0].get(), static_cast<csinn_diso_params*>(params->get()));
        };
        break;
.....
}

void ShlEltwiseExecutor::exec(...) {
    for (size_t i = 0; i < src.size(); i++) {
        srcTensors[i] = srcTensors[i].cloneWithNewShape(src[i]->getDescPtr()->getShape().getStaticDims());
        srcTensors[i].setData(src[i]->getData());
    }
    for (size_t i = 0; i < dst.size(); i++) {
        dstTensors[i] = dstTensors[i].cloneWithNewShape(dst[i]->getDescPtr()->getShape().getStaticDims());
        dstTensors[i].setData(dst[i]->getData());
    }
    OPENVINO_ASSERT(exec_func() == CSINN_TRUE, "Executor failed to execute");
}
```

What do you think? Can we try to so?

Then we don't need to implement some `index_sequence`, `callFunc` and other helpers.

Also then you don't need to save `params` to attributes and not introduce `IShlParams` to save as pointer.

Could you please think about that?

Reply:
If it's possible, you can create local variable `std::function<int()> init_func` and init by the same way. And to the end of `Executor::init()` you can call `OPENVINO_ASSERT(init_func() == CSINN_TRUE, "Executor failed to init");`

Reply:
Actually I used to init all the function pointer in the during the `init` stage before, at that time everything went fine. 

However when the `setShape()` interface is hidden and replaced with `cloneWithNewShape()`, the former method does not work now.  The detailed reasons are as followed:
1. The method I use now will store all the params **statically**, which is used for later function invocation.
2. As I store the smart pointer of `ShlTensor` and `IShlParams` as params, you can set some specific values inside these objects (like `data` or `shape`). In this way the former method still worked, as the content that the smart pointers pointed to still remained the same address.
3. However, with `cloneWithNewShape()` interface, the data that origin pointer points to are no longer useful, so we have to reset the params again in `exec` to ensure the right pointer.

Actually I have been looking for the way to implement a random function pointer that can accept any types of arguments and return any types of value, just like the `func` in following Python code:
```python
def hello_world(param):
    return 1

func = hello_world
```
But I failed to find a way in cpp code, leading me to this little 'ugly' implementation. :(

Reply:
But I think that `index_sequence `, `callFunc` and `params` might be possible to be removed from the class. I will give it a try!

Reply:
`SHLExecutor` for Eltwise ops as `ACLExecutor` has `init` method which is called in `Eltwise::PrepareParams` stage [with executor creation](https://github.com/BHbean/openvino/blob/e70353fdd856913b25f53211a9509bf6bb9e8139/src/plugins/intel_cpu/src/nodes/eltwise.cpp#L2636).
As you know, OpenVINO supports two shape types: static (known on model compilation stage) and dynamic (they will be defined only in inference stage when user set inputs to the model).
If the op has static shapes on model compilation, `Eltwise::PrepareParams` is called in `Eltwise::CreatePrimitive` when all shapes are initialized and known in descriptors. We call them `StaticDims`.
If the op has dynamic shapes on model compilation, `Eltwise::PrepareParams` is called in `Node::updateDynamicParams()` after `ShapeInference` (Shape inference is process of propagation of input shapes of the model to its outputs: each op takes parent output shapes and set to own output by specific rules). It means, that in dynamic case `Eltwise::PrepareParams` also knows the `StaticDims`.
It means that can init `ShlTensors` with `StaticDims` in `ShlExecutor::init`  and no need to update them in `ShlExecutor::exec`. Thus, `ShlExecutor::exec` will update only data pointers, other information in tensors are constant.

Because of that, I think we can try to initialize `exec_func` in  `ShlExecutor::init` method and just update data pointers in tensors and call this `exec_func` in `ShlExecutor::exec` 😃 

What do you think?

Reply:
Thanks for the detailed explanation! If all the shape information can be ensured in `ShlExecutor::init`, I think your implemention is indeed a better way! I will give it a try!

Reply:
The lambda function used here would cause segfault during the test. What about initialzing the `ShlTensor` with static shapes and storing the `exec_func` in the origin way during `init` process?

Reply:
I just found instead of passing `staticDims` to the constructor of `ShlTensor`, using the `cloneWithNewShape()` interface would not cause the segfault problem, and it works now! :)